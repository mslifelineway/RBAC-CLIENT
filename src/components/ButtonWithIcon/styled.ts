import { Button } from "@mui/material";
import styled from "styled-components";

export const StyledButton = styled(Button)(({ theme }) => ({
  "&.MuiButton-root": {
    textTransform: "none",
    color: "#5e5e5e",
    "&.active": {
      color: "#1976d2",
    },
    "&.active-bg": {
      color: "#1976d2",
      backgroundColor: "#cce6ff",
    },
    "&:hover": {
      color: "#1976d2",
    },
  },
}));
