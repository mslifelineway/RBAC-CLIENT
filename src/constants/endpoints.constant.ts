export const endpoints = {
  login: "/login",
  logout: "/logout",
  createAccount: "/",
  getAdmins: "/",
};
