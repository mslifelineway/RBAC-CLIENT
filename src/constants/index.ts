export * from "./path.constant";
export * from "./drawer.constant";
export * from "./endpoints.constant";
export * from "./current-session.constant";
export * from "./toast.constant";
export * from "./message.constant";
