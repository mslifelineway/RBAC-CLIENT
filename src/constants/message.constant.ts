export const messages = {
  accountCreated: "Account Successfully Created.",
  loginSuccess: "Successfully logged in.",
  roleCreated: "Role successfully created.",
  rolesAssigned: "Roles successfully assigned.",
  permissionCreated: "Permission successfully created.",
  permissionsAssigned: "Permissions successfully assigned.",
};
