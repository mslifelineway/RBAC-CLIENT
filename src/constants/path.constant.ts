export const paths = {
  root: "/",
  login: "/login",
  dashboard: "/dashboard",
  employees: "/employees",
  createEmployee: "/employees/create",
};
