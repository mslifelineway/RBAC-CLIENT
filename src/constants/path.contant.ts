export const paths = {
  root: "/",
  login: "/login",
  dashboard: "/dashboard",
};
