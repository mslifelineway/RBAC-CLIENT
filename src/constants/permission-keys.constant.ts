export const permissionKeys = {
  viewDashboard: "WDLFXGNE",
  viewEmployees: "QEDLEQOZ",
  createEmployee: "VTBSXSBW",
  viewRoles: "RIVGPVBT",
  createRole: "KJQQXKBV",
  assignRole: "BRKUJUHH",
  viewPermissions: "PXXKGXOM",
  createPermission: "RIUJQDMQ",
  assignPermission: "PEJQPYMQ",
};
