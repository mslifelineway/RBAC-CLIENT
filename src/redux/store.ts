import { configureStore } from "@reduxjs/toolkit";
import { TypedUseSelectorHook, useSelector } from "react-redux";
import { rootReducer } from "./reducer";
import Cookies from "js-cookie";
import { CURRENT_USER } from "../constants";

const persistedState = Cookies.get(CURRENT_USER)
  ? JSON.parse(Cookies.get(CURRENT_USER)!)
  : null;

export const store = configureStore({
  reducer: rootReducer,
  preloadedState: {
    authReducer: {
      isAuthenticating: false,
      isAuthenticated: Boolean(persistedState),
      error: null,
      data: persistedState,
    },
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export const RequestStatus = {
  FULFILLED: "fulfilled",
  PENDING: "pending",
  REJECTED: "rejected",
};
