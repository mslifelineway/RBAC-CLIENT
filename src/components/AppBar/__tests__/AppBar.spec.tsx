import React from "react";
import { create } from "react-test-renderer";
import { AppBar } from "..";
import { Provider } from "react-redux";
import { store } from "../../../redux/store";
import { IconButton, Typography } from "@mui/material";
import { StyledAppBar } from "../styled";
import { ToastContainer } from "react-toastify";
import { toastOptions } from "../../../constants";

jest.mock("react-router-dom", () => {
  const original = jest.requireActual("react-router-dom");

  return {
    ...original,
    Link: ({ to, children }: { to: string; children: React.ReactNode }) => (
      <a href={to}>{children}</a>
    ),
  };
});

jest.mock("react-dom", () => {
  const original = jest.requireActual("react-dom");

  return {
    ...original,
    createPortal: (node: JSX.Element) => node,
  };
});

describe("AppBar Component", () => {
  test("renders correctly with default props", () => {
    const handleDrawerOpen = jest.fn();
    const component = create(
      <Provider store={store}>
        <ToastContainer {...toastOptions} />
        <AppBar handleDrawerOpen={handleDrawerOpen} />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("renders correctly with open state", () => {
    const handleDrawerOpen = jest.fn();
    const component = create(
      <Provider store={store}>
        <ToastContainer {...toastOptions} />
        <AppBar handleDrawerOpen={handleDrawerOpen} open={true} />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("calls handleDrawerOpen when IconButton is clicked", () => {
    const handleDrawerOpen = jest.fn();
    const component = create(
      <Provider store={store}>
        <ToastContainer {...toastOptions} />
        <AppBar handleDrawerOpen={handleDrawerOpen} />
      </Provider>
    );
    const iconButton = component.root.findByType(IconButton);
    iconButton.props.onClick();
    expect(handleDrawerOpen).toHaveBeenCalled();
  });

  test("renders Typography with text 'RBAC'", () => {
    const handleDrawerOpen = jest.fn();
    const component = create(
      <Provider store={store}>
        <ToastContainer {...toastOptions} />
        <AppBar handleDrawerOpen={handleDrawerOpen} />
      </Provider>
    );
    const typography = component.root.findByType(Typography);
    expect(typography.props.children).toEqual("RBAC");
  });

  test("renders StyledAppBar with position 'fixed'", () => {
    const handleDrawerOpen = jest.fn();
    const component = create(
      <Provider store={store}>
        <ToastContainer {...toastOptions} />
        <AppBar handleDrawerOpen={handleDrawerOpen} />
      </Provider>
    );
    const styledAppBar = component.root.findByType(StyledAppBar);
    expect(styledAppBar.props.position).toEqual("fixed");
  });
});
