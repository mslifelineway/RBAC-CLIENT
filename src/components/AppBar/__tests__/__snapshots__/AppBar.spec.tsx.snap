// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Component renders correctly with default props 1`] = `
Array [
  <div
    className="Toastify"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />,
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-otxe5w-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-p55zm9-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-8u39c-MuiTypography-root"
      >
        RBAC
      </div>
      <div
        className="MuiBox-root css-i9gxme"
      />
      <button
        aria-controls=":r0:"
        aria-expanded={false}
        aria-haspopup="menu"
        className="base-MenuButton-root css-saewb7"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
          data-testid="AccountCircleOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2M7.35 18.5C8.66 17.56 10.26 17 12 17s3.34.56 4.65 1.5c-1.31.94-2.91 1.5-4.65 1.5s-3.34-.56-4.65-1.5m10.79-1.38C16.45 15.8 14.32 15 12 15s-4.45.8-6.14 2.12C4.7 15.73 4 13.95 4 12c0-4.42 3.58-8 8-8s8 3.58 8 8c0 1.95-.7 3.73-1.86 5.12"
          />
          <path
            d="M12 6c-1.93 0-3.5 1.57-3.5 3.5S10.07 13 12 13s3.5-1.57 3.5-3.5S13.93 6 12 6m0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11"
          />
        </svg>
      </button>
      <div
        className="base-Popup-root base-Menu-root"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "top": 0,
            "visibility": "hidden",
          }
        }
      >
        <div
          className="placement-bottom closed"
          onTransitionEnd={[Function]}
        >
          <ul
            className="base-Menu-listbox css-grlzle"
            id=":r0:"
            onBlur={[Function]}
            onKeyDown={[Function]}
            role="menu"
            tabIndex={-1}
          >
            <li
              className="base-MenuItem-root css-v5whyu"
              id=":r1:"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onPointerOver={[Function]}
              role="menuitem"
              tabIndex={-1}
            >
              Profile
            </li>
            <li
              className="base-MenuItem-root css-v5whyu"
              id=":r3:"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onPointerOver={[Function]}
              role="menuitem"
              tabIndex={-1}
            >
              Language settings
            </li>
            <li
              className="base-MenuItem-root css-v5whyu"
              id=":r5:"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onPointerOver={[Function]}
              role="menuitem"
              tabIndex={-1}
            >
              Log out
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>,
]
`;

exports[`AppBar Component renders correctly with open state 1`] = `
Array [
  <div
    className="Toastify"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />,
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-169pbdt-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-189zzm0-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-8u39c-MuiTypography-root"
      >
        RBAC
      </div>
      <div
        className="MuiBox-root css-i9gxme"
      />
      <button
        aria-controls=":r7:"
        aria-expanded={false}
        aria-haspopup="menu"
        className="base-MenuButton-root css-saewb7"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
          data-testid="AccountCircleOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2M7.35 18.5C8.66 17.56 10.26 17 12 17s3.34.56 4.65 1.5c-1.31.94-2.91 1.5-4.65 1.5s-3.34-.56-4.65-1.5m10.79-1.38C16.45 15.8 14.32 15 12 15s-4.45.8-6.14 2.12C4.7 15.73 4 13.95 4 12c0-4.42 3.58-8 8-8s8 3.58 8 8c0 1.95-.7 3.73-1.86 5.12"
          />
          <path
            d="M12 6c-1.93 0-3.5 1.57-3.5 3.5S10.07 13 12 13s3.5-1.57 3.5-3.5S13.93 6 12 6m0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11"
          />
        </svg>
      </button>
      <div
        className="base-Popup-root base-Menu-root"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "top": 0,
            "visibility": "hidden",
          }
        }
      >
        <div
          className="placement-bottom closed"
          onTransitionEnd={[Function]}
        >
          <ul
            className="base-Menu-listbox css-grlzle"
            id=":r7:"
            onBlur={[Function]}
            onKeyDown={[Function]}
            role="menu"
            tabIndex={-1}
          >
            <li
              className="base-MenuItem-root css-v5whyu"
              id=":r8:"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onPointerOver={[Function]}
              role="menuitem"
              tabIndex={-1}
            >
              Profile
            </li>
            <li
              className="base-MenuItem-root css-v5whyu"
              id=":ra:"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onPointerOver={[Function]}
              role="menuitem"
              tabIndex={-1}
            >
              Language settings
            </li>
            <li
              className="base-MenuItem-root css-v5whyu"
              id=":rc:"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onPointerOver={[Function]}
              role="menuitem"
              tabIndex={-1}
            >
              Log out
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>,
]
`;
